tests/src/Kernel/RequirementsTest.php

```diff
@@ -124,9 +125,50 @@ public function testRequirements(): void {
     $this->assertArrayHasKey('collabora_online_settings_wopi_proof', $requirements);
 
     // Change the XML response to contain a proof key.
-    $file = dirname(__DIR__, 2) . '/fixtures/discovery.proof-key.xml';
+    $file_reference = dirname(__DIR__, 2) . '/fixtures/discovery.proof-key.xml';
+    $this->invalidateDiscoveryCache();
+
     $requirements = collabora_online_requirements('runtime');
-    $this->assertEmpty($requirements);
+    $this->assertSame([], $requirements);
+  }
+
+  /**
+   * Creates and registers a mock http client.
+   *
+   * @param string|null $file
+   *   Path to a mock XML file, by reference.
+   *   Can be changed to let the http client return different xml content.
+   */
+  protected function createMockHttpClient(string|null &$file = NULL): void {
+    $original_client = $this->container->get('http_client');
+    $http_client_get = function (...$args) use (&$file, $original_client) {
+      if ($file === NULL) {
+        return $original_client->get(...$args);
+      }
+      $xml = file_get_contents($file);
+      return new Response(
+        200,
+        [],
+        $xml,
+      );
+    };
+    $client = $this->createMock(Client::class);
+    $client->method('get')
+      ->willReturnCallback(
+        function (...$args) use (&$http_client_get): Response {
+          return $http_client_get(...$args);
+        },
+      );
```

Comment:
```suggestion
    $client = $this->createMock(Client::class);
    $client->method('get')
      ->willReturnCallback(
        function (...$args) use (&$file, $original_client): Response {
          if ($file === NULL) {
            return $original_client->get(...$args);
          }
          $xml = file_get_contents($file);
          return new Response(200, [], $xml);
        },
      );
```

Reply:
right. this callback was meant to be replaceable at some point but then that was not needed anymore.
I changed it.